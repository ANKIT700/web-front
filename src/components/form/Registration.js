import React, { useState } from "react";

export default function Registration() {
  const [state, setState] = useState({
    fname: "",
    lname: "",
    email: "",
    password: "",
    mobile: "",
  });

  const handleChange = (e) => {
    const { name, value } = e.target;
    setState({
      ...state,
      [name]: value,
    });
  };
  return (
    <div>
      <form style={{ display: "inline-grid" }}>
        <label>First Name:</label>
        <input
          type="text"
          name="fname"
          onChange={handleChange}
          value={state.fname}
        />
        <label>Last Name:</label>
        <input
          type="text"
          name="lname"
          onChange={handleChange}
          value={state.lname}
        />
        <label>Email:</label>
        <input
          type="text"
          name="email"
          onChange={handleChange}
          value={state.email}
        />
        <label>Mobile Number: </label>
        <input
          type="text"
          name="mobile"
          onChange={handleChange}
          value={state.mobile}
        />
        <label>Password: </label>
        <input
          type="password"
          name="password"
          onChange={handleChange}
          value={state.password}
        />
        <input type="submit" />
      </form>
    </div>
  );
}
