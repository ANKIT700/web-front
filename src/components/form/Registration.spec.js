import { configure, shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import Registration from "./Registration";

configure({ adapter: new Adapter() });

describe("<Registration/>", () => {
  let wrapper = shallow(<Registration />);

  beforeEach(() => {
    wrapper = shallow(<Registration />);
  });
  afterEach(() => {
    wrapper.unmount();
  });
//   it('should render',()=>{
//       expect(wrapper.exists()).toBe(true);
//       })
//   })

it('should render',()=>{

    expect(wrapper.exists()).toBe(true);
})
  it('should render intitial layout',()=>{
    expect(wrapper.getElements()).toMatchSnapshot();
    
  })
});
