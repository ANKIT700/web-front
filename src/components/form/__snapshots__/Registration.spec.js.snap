// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Registration/> should render intitial layout 1`] = `
Array [
  <div>
    <form
      style={
        Object {
          "display": "inline-grid",
        }
      }
    >
      <label>
        First Name:
      </label>
      <input
        name="fname"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label>
        Last Name:
      </label>
      <input
        name="lname"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label>
        Email:
      </label>
      <input
        name="email"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label>
        Mobile Number: 
      </label>
      <input
        name="mobile"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label>
        Password: 
      </label>
      <input
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
      <input
        type="submit"
      />
    </form>
  </div>,
]
`;
